app/components-react/windows/RenameSource.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect, useState, useRef } from 'react';
+import Form, { useForm } from '../shared/inputs/Form';
+import { TextInput } from 'components-react/shared/inputs';
+import { Services } from '../service-provider';
+import { ModalLayout } from '../shared/ModalLayout';
+import { $t } from '../../services/i18n';
+
+export default function RenameSource() {
+  const { SourcesService, WindowsService, EditorCommandsService } = Services;
+
+  const [name, setName] = useState('');
+
+  const form = useForm();
+
+  const options = useRef(WindowsService.getChildWindowQueryParams());
+
+  useEffect(() => {
+    const source = SourcesService.views.getSource(options.current.sourceId);
+    if (!source) return;
+    setName(source.name);
+  }, []);
```

Comment:
What's the point of using an effect to initialize the state?  Why not just:
`const [name, setName] = useState(SourcesService.views.getSource(options.current.sourceId));`

That entire operation shouldn't involve any IPC and will be serviced entirely from the local Vuex store.

Reply:
If the goal is to avoid making that call on every render (not a bad thing to think about), then the best way to handle this free of race conditions would be with `useMemo`:

```TypeScript
const currentName = useMemo(() => SourcesService.views.getSource(options.current.sourceId));
const [name, setName] = useState(currentName);
```

Reply:
yea i was actually thinking about that myself re: initializing with it or not, but yea i think `useMemo` or `useRef` is probably a better fit

Reply:
just read up on it-- apparently the initialState argument of `useState` is only calculated on the first render so should be fine to directly initialize in there